src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -93,6 +94,13 @@ export class RoslynLanguageServer {
     /** The project files previously opened; we hold onto this for the same reason as _solutionFile. */
     private _projectFiles: vscode.Uri[] = new Array<vscode.Uri>();
 
+    /** The diagnostic results from build displayed by VS Code. When live diagnostics are available for a file, these are errors that only build knows about.
+     * When live diagnostics aren't loaded for a file, then these are all of the diagnostics reported by the build.*/
+    private _diagnosticsReportedByBuild: vscode.DiagnosticCollection;
```

Comment:
can you extract all the diagnostics stuff out of this file?  Trying to move to a world where `roslynLanguageServer.ts` doesn't really have feature code in it.